Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm

```diff
@@ -346,7 +346,7 @@ static void configureRequestWithData(NSMutableURLRequest *request, const Resourc
 void ResourceRequest::doUpdatePlatformHTTPBody()
 {
     if (isNull()) {
-        ASSERT(!m_nsRequest);
+        m_nsRequest = nil;
```

Comment:
I assume this releases the request.

Reply:
Yep.  There were a few cases where we have a URL that now converts to null from a non-null request.  In such cases, we want to have the request be null too.